modules/ding_campaign_plus/modules/ding_campaign_plus_search/ding_campaign_plus_search.module

```diff
@@ -0,0 +1,307 @@
+  return array(
+    '#type' => 'checkbox',
+    '#title' => t('Enable'),
+    '#description' => t('Activate search trigger based on matching subjects from content.'),
+    '#default_value' => empty($default) ? FALSE : $default,
+  );
+}
+
+/**
+ * The node edit form callback defined in the hook_ding_campaign_plus_info.
+ */
+function ding_campaign_plus_search_admin_form(&$form_state) {
+  $triggers = array();
+
+  $values = array();
+  if (isset($form_state['values']['tabs'][DING_CAMPAIGN_PLUS_SEARCH_TYPE]['rules'])) {
+    $values = $form_state['values']['tabs'][DING_CAMPAIGN_PLUS_SEARCH_TYPE]['rules'];
+  }
+
+  // Get campaign rule from node.
+  if (empty($values) && isset($form_state['node']->ding_campaign_plus)) {
+    $values = _ding_campaign_plus_search_get_rules($form_state['node']->nid, $form_state['node']->ding_campaign_plus);
+  }
+
+  $triggers['percent'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Percent of the results'),
+    '#size' => 4,
+    '#description' => t('The percent of the results in the search result.'),
+    '#default_value' => isset($values['percent']) ? $values['percent'] : '30',
+  );
+
+  $triggers['query'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Search query'),
+    '#description' => t('Search query that should trigger the campaign.'),
+    '#default_value' => isset($values['query']) ? $values['query'] : '',
+    '#maxlength' => 2048
+  );
+
+  return $triggers;
+}
+
+/**
+ * Find "facet" rules in the current rules for a given node.
+ *
+ * @param int $nid
+ *   Node id for the node (used as index in static cache).
+ * @param array $rules
+ *   All the rules loaded for the node.
+ *
+ * @return array
+ *   The rules of the basic types.
+ */
+function _ding_campaign_plus_search_get_rules($nid, $rules) {
+  $filtered = drupal_static(__FUNCTION__, array());
+
+  if (!isset($filtered[$nid])) {
+    $filtered[$nid] = array();
+
+    foreach ($rules as $rule) {
+      if ($rule->type == 'search') {
+        $filtered[$nid]  = array(
+          'percent' => $rule->percent,
+          'query' => $rule->query,
+        );
+      }
+    }
+  }
+
+  return $filtered[$nid];
+}
+
+/**
+ * Helper function to look-up rules in the database.
+ *
+ * @param array $keys
+ *   Search keys entered by the user.
+ *
+ * @return array
+ *   Campaign ID's for triggered campaigns.
+ */
+function _ding_campaign_plus_search_match($keys) {
+  $nids = array();
+
+  // Find rules that will match facet.
+  $triggers = db_select('ding_campaign_plus_search', 'dcps')
+    ->fields('dcps')
+    ->execute();
+
+  if ($triggers) {
+    // Run user search to get number of results used to percentage
+    // calculations.
+    $facet_array = array();
+    $conditions = ting_search_conditions_callback($keys);
+    $query = '(' . _ting_search_quote($keys) . ')';
+
+    // Extend query with selected facets.
+    if (isset($conditions['facets']) && $conditions['facets'] != NULL) {
```

Comment:
`$conditions['facets'] != NULL` is superfluous after `isset($conditions['facets'])`.

Reply:
Fixed